src/asmcnc/apps/systemTools_app/screens/calibration/screen_serial_numbers.py

```diff
@@ -403,6 +407,22 @@ def report_info_back_to_user_and_return(self, dt):
             else:
                 self.error_label.text = "Success!!"
 
+    def send_public_keys(self):
+        try:
+            cmd = 'cat ../../.ssh/id_rsa.pub'
+
+            output = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
+            response = output.communicate()[0]
+
+            if len(response) == 0:
+                PopupNoSSHFile()
+                return
+
+            self.calibration_db.send_ssh_keys(self.console_serial_input.text, response)
+        except:
+            log("Couldn't find public key file")
+            PopupNoSSHFile()
```

Comment:
Won't this also fail if WiFi connection is poor or if the DB server is down? Can error messages please be updated accordingly because otherwise (e.g. if the WiFi drops out in future) I will definitely forget and get confused when I can find a perfectly valid key file 😅

Also, what happens if the key file is there but there's nothing in it? (e.g. someone (me) has been checking the key file and they've accidentally deleted the contents in the process 😳)